src/GeckofxHtmlToPDFComponent.cs

```diff
@@ -446,6 +462,158 @@ public void OnSecurityChange(nsIWebProgress aWebProgress, nsIRequest aRequest, u
+			var newTempFile = GetTemporaryPdfFilename(); 
+			var info = new ProcessStartInfo
+			{
+				FileName = "gs",	// if installed on Linux, has simple name and will be in the path
+				Arguments = String.Format("-sDEVICE=pdfwrite -dBATCH -dNOPAUSE -dCompatibilityLevel=1.4 -dDownsampleColorImages=true -dColorImageResolution=600 -sOutputFile=\"{0}\" \"{1}\"",
+					newTempFile, _pathToTempPdf),
+				UseShellExecute = false,		// false for redirection
+				RedirectStandardOutput = true	// redirect for event handler reading asynchronously
+			};
+
+			try
+			{
+				Process gsProcess = new Process();
+				gsProcess.StartInfo = info;
+				if (Environment.OSVersion.Platform == PlatformID.Win32NT)
+					gsProcess.StartInfo.FileName = FindGhostcriptOnWindows();
+				gsProcess.OutputDataReceived += ShrinkOutputHandler;
+				gsProcess.Start();
+				gsProcess.BeginOutputReadLine();	// get asynchronous output
+
+				gsProcess.WaitForExit(600000);		// timeout - 10 minutes? 1 minutes? 5 minutes? 30 sec?
+				gsProcess.OutputDataReceived -= ShrinkOutputHandler;
+
+				if (File.Exists(newTempFile) && gsProcess.ExitCode == 0)
+				{
+					var oldInfo = new FileInfo(_pathToTempPdf);
+					var newInfo = new FileInfo(newTempFile);
+					// If the process actually made the file larger, ignore the result (and delete it).
+					if (newInfo.Length < oldInfo.Length)
+						return newTempFile;
+					File.Delete(newTempFile);
+				}
+			}
+			catch (Exception ex)
+			{
+				// If something bad happens, well, that means we can't shrink the file.  Keep going.
+				// On Windows, we should tell the user about downloading ghostscript.
+				// Except that this gets way too intrusive quickly.  We need to tell the user at least
+				// once, but how do we control (not) repeating the message?
+				//if (info.FileName == "gswin32.exe")
+				//{
+				//	var msg = "Cannot shrink the PDF because ghostscript cannot be found on this computer.  It can be downloaded from https://ghostscript.com/download/.";
+				//	var header = "Cannot shrink PDF";
+				//	MessageBox.Show(msg, header, MessageBoxButtons.OK, MessageBoxIcon.Information);
+				//}
+			}
+			return null;
+		}
+
+		/// <summary>
+		/// Try to find ghostscript on the Windows system, looking in the likeliest installation
+		/// locations and trying to not depend on the exact version installed.
+		/// </summary>
+		/// <remarks>
+		/// Ghostscript 9.21 for 64-bit Windows 10 installs by default as
+		/// C:\Program Files\gs\gs9.21\bin\gswin64c.exe or C:\Program Files\gs\gs9.21\bin\gswin64.exe.
+		/// The former uses the current console window, the latter pops up its own command window.
+		/// </remarks>
+		private string FindGhostcriptOnWindows()
+		{
+			var baseName = "gswin32";
+			if (Environment.Is64BitOperatingSystem)
+				baseName = "gswin64";
+			// Look for the four most common installation locations.
+			var baseDir = @"C:\Program Files\gs";
```

Comment:
Environment.SpecialFolder.ProgramFiles and Environment.SpecialFolder.ProgramFiles86